### ISSUE TITLE
When deploying a Guaranteed QoS in k8s, Envoy occasionally restarts

### ISSUE THREAD

**Issue by hushuai1559** [emissary-ingress/emissary#5812]
my k8s version is 1.28.9，I tried using Envoy versions from 1.17.4 to 1.27.1, but encountered the error "Caught Segmentation fault" in all cases.
The detailed error is as follows:
[2024-12-10 02:10:43.598][180][critical][backtrace] [./source/server/backtrace.h:104] Caught Segmentation fault, suspect faulting address 0x0
2024-12-10T10:10:43.599081709+08:00 [2024-12-10 02:10:43.599][180][critical][backtrace] [./source/server/backtrace.h:91] Backtrace (use tools/stack_decode.py to get line numbers):
2024-12-10T10:10:43.599087228+08:00 [2024-12-10 02:10:43.599][180][critical][backtrace] [./source/server/backtrace.h:92] Envoy version: 6637fd1bab315774420f3c3d97488fedb7fc710f/1.27.2/Clean/RELEASE/BoringSSL
[2024-12-10 02:10:43.599][180][critical][backtrace] [./source/server/backtrace.h:98] #0: [0x7f27578f7320]
2024-12-10T10:10:43.599445995+08:00 [2024-12-10 02:10:43.599][180][critical][backtrace] [./source/server/backtrace.h:98] #1: [0x556b85fa1888]
2024-12-10T10:10:43.599467253+08:00 [2024-12-10 02:10:43.599][180][critical][backtrace] [./source/server/backtrace.h:98] #2: [0x556b86f81087]
2024-12-10T10:10:43.599491208+08:00 [2024-12-10 02:10:43.599][180][critical][backtrace] [./source/server/backtrace.h:98] #3: [0x556b879b487a]
[2024-12-10 02:10:43.599][180][critical][backtrace] [./source/server/backtrace.h:98] #4: [0x556b879b3491]
2024-12-10T10:10:43.599539289+08:00 [2024-12-10 02:10:43.599][180][critical][backtrace] [./source/server/backtrace.h:98] #5: [0x556b85fc4b2b]
2024-12-10T10:10:43.599554492+08:00 [2024-12-10 02:10:43.599][180][critical][backtrace] [./source/server/backtrace.h:98] #6: [0x556b879e9563]
[2024-12-10 02:10:43.599][180][critical][backtrace] [./source/server/backtrace.h:98] #7: [0x7f275794249a]

 I noticed that Envoy retrieves the number of cores from the host machine when starting, which may conflict with Kubernetes' Guaranteed QoS. Could this be the cause of the issue?
 I might be able to control the number of worker threads at startup by setting --concurrency, but logically, it would still conflict with Kubernetes.

**Comment by cindymullins-dw** [comment]
Hi @hushuai1559, this looks like an Envoy question rather than Emissary-ingress.  Can you pls clarify?

**Comment by hushuai1559** [comment]
yes,I have moved on to the issues with Envoy, this can be closed.

**Comment by hushuai1559** [comment]
> Hi @hushuai1559, this looks like an Envoy question rather than Emissary-ingress. Can you pls clarify?

This issue is likely related to Emissary Ingress. In the same scenario, Emissary Ingress version 2.1.2 starts up normally, but after upgrading to version 2.2.2, a Segmentation fault in Envoy causes Emissary Ingress to fail to start. When I use the latest version, 3.9.1, the same problem occurs. Additionally, when I replace the Envoy in version 3.9.1 with the 1.17.4-dev version of Envoy (which is the version used by Emissary Ingress 2.1.2), it still fails to start eventually. Since Emissary Ingress's Envoy is started based on the xDS approach, it is difficult to rule out Emissary Ingress as the root cause. Moreover, when I start Envoy in a static configuration mode on the same Kubernetes cluster, no errors occur.

**Comment by hushuai1559** [comment]
By using the following command to create a Minikube cluster on a Linux machine and then deploying Emissary Ingress 3.9.1 via the Helm method recommended by the official documentation, you will encounter the same error as described.

minikube start --force --base-image='registry.cn-hangzhou.aliyuncs.com/google_containers/kicbase:v0.0.45' \
--kubernetes-version=v1.28.9 \
--container-runtime=containerd \
--extra-config=kubelet.kube-reserved='cpu=0.5,memory=512Mi' \
--extra-config=kubelet.cpu-manager-policy='static' \
--extra-config=kubelet.topology-manager-policy='single-numa-node'
